FormidableLabs/react-native-app-auth
# Adding support for specifying what browser needs to be used for authorization

This fixes #588 and #643 

## Description


This pull request adds the ability to control what browser is used for the authorization process. This helps in complex flows where the authorization is going through a browser and has a sidestep to a third party app.

In our case we were encountering issues, because third party apps use the default browser for return urls, but the in-app browser (Such as `aswebauthenticationsession`) doesn't share cookies with Safari.
By providing these parameters, the developers can fine-tune the login flow for their needs.


**Question for the maintainers:**
I'm not an iOS developer, and am seeing one flaw in the user flow when using an external browser:
If the user cancels the flow by pressing the "back" navigation on the top left of the browser, the promise isn't rejected. Do you have any idea how i can fix this? On Android, the promise is rejected when returning (which is expected)


## Steps to verify


On iOS: 
 - Adding `iosCustomBrowser` with the value `safari` should open safari, instead of a browser modal. 
 - Without providing `iosCustomBrowser`, the behavior should remain the same as before this pull request

On Android:
 - Adding `androidAllowCustomBrowsers` with the value ['chrome'] should open chrome, instead of the in-app browser
 - Without providing `androidAllowCustomBrowsers`, the behavior should remain the same as before this pull request
